x-pack/metricbeat/magefile.go

```diff
@@ -252,6 +272,18 @@ func GoIntegTest(ctx context.Context) error {
 // Use TEST_TAGS=tag1,tag2 to add additional build tags.
 // Use MODULE=module to run only tests for `module`.
 func GoFIPSOnlyIntegTest(ctx context.Context) error {
+	// build integration test binary with otel sub command
+	devtools.BuildSystemTestOTelBinary()
+	args := devtools.DefaultGoTestIntegrationFromHostArgs()
+	// ES_USER must be admin in order for the Go Integration tests to function because they require
+	// indices:data/read/search
+	args.Env["ES_USER"] = args.Env["ES_SUPERUSER_USER"]
+	args.Env["ES_PASS"] = args.Env["ES_SUPERUSER_PASS"]
+	// run integration test from home directory
+	args.Packages = []string{"./tests/integration/"}
+	devtools.GoIntegTestFromHost(ctx, args)
+
+	// run module integration test
```

Comment:
I know trying to avoid any duplication can be unnecessary, but this looks so much like the code for `GoIntegTest`. Perhaps it'd be possible to try to have a generic code that does `GoIntegTest` and the FIPS does only the FIPS specific setup